core/src/main/scala/com/databricks/labs/remorph/RootTableIdentifier.scala

```diff
@@ -0,0 +1,51 @@
+package com.databricks.labs.remorph
+
+import com.databricks.labs.remorph.parsers.{intermediate => ir}
+import com.databricks.labs.remorph.transpilers.TranspilerFactory
+
+class RootTableIdentifier(engine: String) {
+
+  private val transpiler = TranspilerFactory.getTranspiler(engine)
+
+  def processQuery(sql: String, dag: DAG): DAG = {
+    val parsedSql = transpiler.parse(sql)
+    findTables(dag, parsedSql)
+
+  }
+
+  private def getTablName(p: ir.LogicalPlan): Set[String] = {
+    p.find(_.isInstanceOf[ir.NamedTable]).map(_.asInstanceOf[ir.NamedTable].unparsed_identifier).toSet
+  }
+
+  private def findTables(dag: DAG, plan: ir.LogicalPlan): DAG = {
+    var child = ""
+    var parent = Set.empty[String]
+
+    def collectTables(node: ir.LogicalPlan): Unit = {
+      node match {
+        case n: ir.InsertIntoTable =>
```

Comment:
```suggestion
        case ir.NamedTable(name, _, _) =>
```
why do you need so many unnecessary branches?..

Reply:
Tackled